### PR TITLE
feat(idam-sandbox): add idam-web-public-aks.sandbox.platform.hmcts.net

### PR DESCRIPTION
**Pull request by henrydobson**
### Change description ###

Add idam-sandbox
* idam-web-public-aks.sandbox.platform.hmcts.net

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: environments/sbox.tfvars [hunk 1]
```diff
@@ -6,9 +6,9 @@ enable_ssl                 = true
 ssl_mode                   = "AzureKeyVault"
 certificate_key_vault_name = "cftapps-sbox"
 
-shutter_storage = "default"
-cdn_sku         = "Standard_Verizon"
-shutter_rg      = "shutter-app-sbox-rg"
+shutter_storage     = "default"
+cdn_sku             = "Standard_Verizon"
+resource_group_name = "shutter-app-sbox-rg"
```
Comment (timja) [inline environments/sbox.tfvars]: why this change?

File: environments/sbox.tfvars [hunk 2]
```diff
@@ -21,4 +21,22 @@ frontends = [
     certificate_name = "STAR-sandbox-platform-hmcts-net"
     disabled_rules   = {}
   },
+  {
+    name             = "idam-web-public"
+    custom_domain    = "idam-web-public-aks.sandbox.platform.hmcts.net"
+    backend_domain   = ["firewall-sbox-int-palo-sbox.uksouth.cloudapp.azure.com"]
+    certificate_name = "STAR-sandbox-platform-hmcts-net"
+    disabled_rules = {
```
Comment (timja) [inline environments/sbox.tfvars]: prefer exclusions over disabled rules:
https://github.com/hmcts/azure-platform-terraform/blob/master/environments/prod.tfvars#L22-L42

File: environments/sbox.tfvars [hunk 3]
```diff
@@ -6,9 +6,9 @@ enable_ssl                 = true
 ssl_mode                   = "AzureKeyVault"
 certificate_key_vault_name = "cftapps-sbox"
 
-shutter_storage = "default"
-cdn_sku         = "Standard_Verizon"
-shutter_rg      = "shutter-app-sbox-rg"
+shutter_storage     = "default"
```
Comment (naikajah) [inline environments/sbox.tfvars]: we no longer use this. You can remove this line from sbox.tfvars.

File: environments/sbox.tfvars [hunk 4]
```diff
@@ -21,4 +21,22 @@ frontends = [
     certificate_name = "STAR-sandbox-platform-hmcts-net"
     disabled_rules   = {}
   },
+  {
+    name             = "idam-web-public"
+    custom_domain    = "idam-web-public-aks.sandbox.platform.hmcts.net"
```
Comment (timja) [inline environments/sbox.tfvars]: there needs to be a PR to azure-public-dns to set up the domain validation,

https://github.com/hmcts/azure-public-dns/blob/master/environments/sandbox.yml#L52-L58

and also a cdnverify for the shuttering:
https://github.com/hmcts/azure-public-dns/blob/master/environments/prod/track-benefit-appeal-service-gov-uk.yml#L30-L32

File: environments/sbox.tfvars (hunk 1 above)
Comment (naikajah) [inline environments/sbox.tfvars]: Why is the change in variable name?
This should be 
```
shutter_rg      = "shutter-app-sbox-rg"
```

Comment (henrydobson) [reply environments/sbox.tfvars]: MC error. Didn't notice that.

Comment (henrydobson) [reply environments/sbox.tfvars]:
```suggestion
shutter_rg      = "shutter-app-sbox-rg"
```

File: environments/sbox.tfvars (hunk 3 above)
Comment (henrydobson) [inline environments/sbox.tfvars]:
```suggestion
shutter_storage = "default"
```

File: environments/sbox.tfvars [hunk 5]
```diff
@@ -6,9 +6,9 @@ enable_ssl                 = true
 ssl_mode                   = "AzureKeyVault"
 certificate_key_vault_name = "cftapps-sbox"
 
-shutter_storage = "default"
-cdn_sku         = "Standard_Verizon"
-shutter_rg      = "shutter-app-sbox-rg"
+shutter_storage     = "default"
+cdn_sku             = "Standard_Verizon"
```
Comment (henrydobson) [inline environments/sbox.tfvars]:
```suggestion
cdn_sku         = "Standard_Verizon"
```

File: environments/sbox.tfvars (hunk 1 above)
Comment (henrydobson) [reply environments/sbox.tfvars]: https://github.com/hmcts/azure-platform-terraform/pull/97/files/af53a8fd9c65bd1b888c62c1da5f24d91538a19e#r376359725

File: environments/sbox.tfvars (hunk 4 above)
Comment (henrydobson) [reply environments/sbox.tfvars]: Not exactly sure how you need this. Can you provide any more instructions?

Comment (timja) [reply environments/sbox.tfvars]: Azure front door and the azure cdn we use for shuttering both require domain ownership proof, we always use the indirect validation because otherwise if someone shutters an app we can’t deploy changes to front door as it checks that you own the domain every time you make changes to it. 

Is there a specific bit you aren’t sure about?

**Comment by timja** [comment]
Needs re-submitting from origin, I've added CODEOWNERS and adjusted permissions
